### PR TITLE
Fix tests broken by No-Op changes

### PR DESCRIPTION
**Pull request by nkolev92**
Fixing the old tests that got broken by the source plumbing in changes.  There's initial VS work done here too, where we set the globalpackages/fallback folders.

### CODE REVIEW

File: src/NuGet.Core/NuGet.Build.Tasks/GetRestoreDotnetCliToolsTask.cs [hunk 1]
```diff
@@ -33,20 +33,25 @@ public class GetRestoreDotnetCliToolsTask : Task
         public string ToolFramework { get; set; }
 
         /// <summary>
-        /// NuGet sources, ; delimited
+        /// NuGet sources
         /// </summary>
-        public string RestoreSources { get; set; }
+        public string[] RestoreSources { get; set; }
```
Comment (emgarten) [inline src/NuGet.Core/NuGet.Build.Tasks/GetRestoreDotnetCliToolsTask.cs:38]: 🥇

File: src/NuGet.Core/NuGet.Build.Tasks/GetRestoreSettingsTask.cs [hunk 2]
```diff
@@ -132,17 +130,12 @@ public override bool Execute()
...
-                log.LogDebug($"(in) RestoreDirectory '{RestoreDirectory}'");
-            }
-
             if (RestoreSolutionDirectory != null)
             {
                 log.LogDebug($"(in) RestoreSolutionDirectory '{RestoreSolutionDirectory}'");
             }
 
-            var settings = ReadSettings(RestoreSolutionDirectory, string.IsNullOrEmpty(RestoreDirectory) ? Path.GetDirectoryName(ProjectUniqueName) : RestoreDirectory, RestoreConfigFile);
+            var settings = ReadSettings(RestoreSolutionDirectory, Path.GetDirectoryName(ProjectUniqueName) , RestoreConfigFile);
```
Comment (emgarten) [inline src/NuGet.Core/NuGet.Build.Tasks/GetRestoreSettingsTask.cs:138]: ProjectUniqueName doesn't have to be a path (it usually is). 

This task should also take ``ProjectPath`` and use that, kind of redundant, but other tasks do it this way to allow project unique name to change.

File: src/NuGet.Core/NuGet.PackageManagement/BuildIntegration/DependencyGraphRestoreUtility.cs [hunk 3]
```diff
@@ -338,6 +353,9 @@ public static class DependencyGraphRestoreUtility
 
                 foreach (var packageSpec in packageSpecs)
                 {
+                    packageSpec.RestoreMetadata.FallbackFolders = (IList<string>)fallbackFolders;
```
Comment (emgarten) [inline src/NuGet.Core/NuGet.PackageManagement/BuildIntegration/DependencyGraphRestoreUtility.cs]: nit: AsList

File: src/NuGet.Core/NuGet.PackageManagement/BuildIntegration/DependencyGraphRestoreUtility.cs [hunk 4]
```diff
@@ -302,23 +308,32 @@ public static class DependencyGraphRestoreUtility
...
-            SettingsUtility.GetFallbackPackageFolders(context.Settings);
-            
+            var globalPackagesFolder = SettingsUtility.GetGlobalPackagesFolder(context.Settings);
+            var fallbackFolders = SettingsUtility.GetFallbackPackageFolders(context.Settings);
 
             foreach (var packageSpec in context.DeferredPackageSpecs)
             {
+                //TODO NK - Does this really make sense? Anything unforeseen here? 
+                packageSpec.RestoreMetadata.FallbackFolders = packageSpec.RestoreMetadata.FallbackFolders ?? fallbackFolders.AsList();
+                packageSpec.RestoreMetadata.PackagesPath = packageSpec.RestoreMetadata.PackagesPath ?? globalPackagesFolder;
```
Comment (emgarten) [inline src/NuGet.Core/NuGet.PackageManagement/BuildIntegration/DependencyGraphRestoreUtility.cs]: I would force override these for now

File: test/NuGet.Core.FuncTests/NuGet.XPlat.FuncTest/XPlatTestUtils.cs [hunk 5]
```diff
@@ -180,12 +182,18 @@ public static XmlReaderSettings CreateSafeSettings(bool ignoreWhiteSpace = false
...
             string projectFrameworks)
         {
+            var settings = Settings.LoadDefaultSettings(Path.GetDirectoryName(pathContext.NuGetConfig), Path.GetFileName(pathContext.NuGetConfig), null);
             var project = SimpleTestProjectContext.CreateNETCoreWithSDK(
                     projectName: projectName,
                     solutionRoot: pathContext.SolutionRoot,
                     isToolingVersion15: true,
                     frameworks: MSBuildStringUtility.Split(projectFrameworks));
 
+            project.FallbackFolders = (IList<string>) SettingsUtility.GetFallbackPackageFolders(settings);
```
Comment (emgarten) [inline test/NuGet.Core.FuncTests/NuGet.XPlat.FuncTest/XPlatTestUtils.cs:192]: nit: aslist

File: test/NuGet.Core.Tests/NuGet.Commands.Test/RequestFactoryTests.cs [hunk 6]
```diff
@@ -64,24 +65,19 @@ public void RequestFactory_RestorePackagesArgRelativeToCwd()
...
         [Fact]
-        public async Task RequestFactory_FindProjectJsonFilesInDirectory()
+        public async Task RequestFactory_ReadDGSpec()
         {
             // Arrange
             var cache = new RestoreCommandProvidersCache();
-            var provider = new ProjectJsonRestoreRequestProvider(cache);
-
+            var provider = new DependencyGraphFileRequestProvider(cache);
+            //Debugger.Launch();
```
Comment (emgarten) [inline test/NuGet.Core.Tests/NuGet.Commands.Test/RequestFactoryTests.cs]: nit: remove

**Review by emgarten — APPROVED** [review]

File: src/NuGet.Core/NuGet.Build.Tasks/GetRestoreSettingsTask.cs (hunk 2 above)
Comment (nkolev92) [reply src/NuGet.Core/NuGet.Build.Tasks/GetRestoreSettingsTask.cs:138]: Will do it later. 
Updated the Issue with this comment:
https://github.com/NuGet/Home/issues/5181

**Pull request merged**